components/brave_wallet_ui/components/shared/solana-transaction-instruction/solana-transaction-instruction.tsx

```diff
@@ -36,115 +36,183 @@ import {
-            </TransactionTitle>
-          </SectionRow>
-          {Object.keys(params).length > 0 && (
-            <>
-              <Divider />
-              {
-                Object.entries(params).map(([key, value]) => {
-                  const paramName = getSolanaInstructionParamKeyName(key as SolanaInstructionParamKeys)
-                  const isAddressParam = key.toString().toLowerCase().includes('pubkey')
-                  const formattedParamValue = (key as SolanaInstructionParamKeys === 'lamports'
-                    // format lamports to SOL
-                    ? new Amount(value.toString()).div(LAMPORTS_PER_SOL).formatAsAsset(9, 'SOL')
-
-                    // show friendly account address names
-                    : isAddressParam ? findAccountName(accounts, value.toString()) ?? value
-
-                      // unformatted param value
-                      : value
-                  ).toString()
-
-                  return <InstructionParamBox key={key}>
-                    <var>{paramName}</var>
-                    {isAddressParam
-                      ? <Tooltip
-                        isAddress
-                        text={value.toString()}
-                        position='left'
-                      >
-                        <AddressText>{formattedParamValue}</AddressText>
-                      </Tooltip>
-                      : <samp>{formattedParamValue}</samp>
-                    }
-                  </InstructionParamBox>
-                })
+          <CodeSectionTitle>
+            {getLocale('braveWalletSolanaAccounts')}
+          </CodeSectionTitle>
+          {accountParams.map(({ localizedName, name, value }) => {
+            // signers param
+            if (name === BraveWallet.SIGNERS) {
+              if (!value) {
+                return null
+              }
+
+              const signers = value.split(',')
+              if (!signers.length) {
+                return null
               }
-            </>
-          )}
+
+              return (
+                <InstructionParamBox key={name}>
+                  <var>{localizedName}</var>
+                  {signers.map((pubkey) => {
+                    return (
+                      <AddressParamValue
+                        key={pubkey}
+                        accounts={accounts}
+                        pubkey={pubkey}
+                      />
+                    )
+                  })}
+                </InstructionParamBox>
+              )
+            }
+
+            // other account params
+            return (
+              <InstructionParamBox key={name}>
+                <var>{localizedName}</var>
+                <AddressParamValue accounts={accounts} pubkey={value} />
+              </InstructionParamBox>
+            )
+          })}
         </>
       )
-      }
-    </InstructionBox >
+    )
+  }, [accountParams])
+
+  // render
+  if (!type) {
+    // instruction not decodable by backend
```

Comment:
Please help double check that we would list the pubkeys in account_metas in this case, thanks.

Reply:
updated https://github.com/brave/brave-core/compare/d5cb55b4ac67e9cbf9fc439433cb8b56b0b30152..caa3600de0ebc1e09a300fc46e34ca1ee3a150ea